order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -70,10 +86,161 @@
 			return new ResponseEntity<>(new ApiErrorsWS(errorWSList), HttpStatus.UNPROCESSABLE_ENTITY);
 		}
 
-//		OrderBean orderBean = orderMapper.toBean(orderWS);
-//		OrderBean order = orderService.createOrder(orderBean);
+		OrderBean orderBean = orderMapper.toBean(orderWS, new CycleAvoidingMappingContext());
+
+		OrderBean order = orderService.createOrder(orderBean);
+
+		return new ResponseEntity<>(orderMapper.toWS(order, new CycleAvoidingMappingContext()), HttpStatus.OK);
```

Comment:
Anyway to hide `new CycleAvoidingMappingContext()` inside `xxxMaper` so that we don't need to set it every time?

Reply:
I tried it, trying to override it in mapstruct but can't work. I will try to find another solution

Reply:
This is still not working even I tried to overloading it